tests/unit/test_errorhandler.py

```diff
@@ -10,101 +10,74 @@
-        try:
-            handler(http_response, response, operation)
-        except errorhandler.ServerError as e:
-            # First, the operation name should be in the error message.
-            self.assertIn('OperationName', str(e))
-            # We should state that this is a ServerError.
-            self.assertIn('server error', str(e))
-            # And these values should be available on the exception
-            # so clients can access this information programmatically.
-            self.assertEqual(e.error_code, 'InternalError')
-            self.assertEqual(e.error_message, 'An internal error has occurred')
-            self.assertEqual(e.operation_name, 'OperationName')
-        except Exception as e:
-            self.fail("Unexpected error raised: %s" % e)
-        else:
-            self.fail("Expected errorhandler.ServerError to be raised "
-                      "but no exception was raised.")
-
-    def test_no_exception_raised_on_200(self):
-        response = {
-            'CommonPrefixes': [],
-            'Contents': [],
-        }
-        handler = errorhandler.ErrorHandler()
-        http_response = self.create_http_response(status_code=200)
-        # We're manually using the try/except form because
-        # we want to catch the exception and assert that it has specific
-        # attributes on it.
-        operation = mock.Mock()
-        operation.name = 'OperationName'
-        try:
-            self.assertIsNone(handler(http_response, response, operation))
-        except errorhandler.BaseOperationError as e:
-            self.fail("Unexpected error raised: %s" % e)
-
+            cr = self.error_handlers_chain.handle_exception(e, stdout, stderr)
+            self.assertEqual(cr, case.rc, case.exception.__class__)
+            self.assertIn(case.message, stderr.getvalue())
+            self.assertEqual('', stdout.getvalue())
 
-if __name__ == '__main__':
-    unittest.main()
+    def test_error_handling(self):
+        Case = namedtuple('Case', [
+            'exception',
+            'rc',
+            'message',
+        ])
+        cases = [
+            Case(Exception(), 255, '\n\n'),
+            Case(NoRegionError(), 253, 'region'),
+            Case(NoCredentialsError(), 253, 'credentials'),
+            Case(ClientError(error_response={}, operation_name=''), 254,
+                 'An error occurred'
+            ),
+            Case(BotocoreParamValidationError(report='param_name'), 252,
+                 'param_name'),
+            Case(UnknownArgumentError(), 252, ''),
+            Case(ArgParseException(), 252, '\n\n'),
+            Case(ParamSyntaxError(), 252, '\n\n'),
+            Case(ParamError(cli_name='cli', message='message'), 252,
+                 "'cli': message"),
+            Case(ParamValidationError(), 252, '\n\n'),
+            Case(ConfigurationError(), 253, '\n\n'),
+        ]
+        for case in cases:
+            self._assert_rc_and_error_message(case)
```

Comment:
So it may make sense to use `yield` here. Similar to this test case: https://github.com/aws/aws-cli/blob/345da7177703ba297fd4b192bd8e9ad04898a592/tests/functional/autoprompt/test_autoprompt.py#L66. Mainly yield allows us to get through all of the test cases and not have it short circuit on us if one fails and potentially hide other failures.